pkg/logql/syntax/ast.go

```diff
@@ -283,6 +284,17 @@ func newLineFilterExpr(ty labels.MatchType, op, match string) *LineFilterExpr {
 	}
 }
 
+func newOrLineFilter(left, right *LineFilterExpr) *LineFilterExpr {
+	if left.Ty == labels.MatchEqual || left.Ty == labels.MatchRegexp {
+		left.Or = right
+		return left
+	}
+
+	// !(left or right) == (!left and !right).
+	right.Ty = left.Ty
```

Comment:
What is `Ty`?

Reply:
The matcher type `MatchEqual`, `MatchNotEqual` etc.